crates/payload/basic/src/lib.rs

```diff
@@ -561,13 +608,14 @@ fn build_payload<Pool, Client>(
         Client: StateProviderFactory,
         Pool: TransactionPool,
     {
+        let extra_data = config.extra_data();
         let PayloadConfig {
             initialized_block_env,
             initialized_cfg,
             parent_block,
-            extra_data,
             attributes,
             chain_spec,
+            ..
```

Comment:
with https://github.com/paradigmxyz/reth/pull/4153

the actual build function can be configured, so I'd recommend duplicating this function entirely with the new trait

requires one smol additional reth change though